drivers/media/i2c/ov5693.c

```diff
@@ -1052,103 +1023,118 @@ static int ov5693_init(struct v4l2_subdev *sd)
 	}
 
 	/*change initial focus value for ad5823*/
-	if (dev->vcm == VCM_AD5823) {
-		dev->focus = AD5823_INIT_FOCUS_POS;
+	if (ov5693->vcm == VCM_AD5823) {
+		ov5693->focus = AD5823_INIT_FOCUS_POS;
 		ov5693_t_focus_abs(sd, AD5823_INIT_FOCUS_POS);
 	} else {
-		dev->focus = 0;
+		ov5693->focus = 0;
 		ov5693_t_focus_abs(sd, 0);
 	}
 
-	ov5693_isp_configure(dev);
-	mutex_unlock(&dev->input_lock);
+	ov5693_isp_configure(ov5693);
+	mutex_unlock(&ov5693->lock);
 
 	return 0;
 }
 
-static int __power_up(struct v4l2_subdev *sd)
+static int ov5693_sw_standby(struct ov5693_device *ov5693, bool standby)
 {
-	struct i2c_client *client = v4l2_get_subdevdata(sd);
-	struct ov5693_device *sensor = to_ov5693_sensor(sd);
 	int ret;
 
-	ret = clk_prepare_enable(sensor->clk);
-	if (ret) {
-		dev_err(&client->dev, "Error enabling clock\n");
-		return -EINVAL;
-	}
-
-	if (sensor->indicator_led)
-		gpiod_set_value_cansleep(sensor->indicator_led, 1);
+	ret = ov5693_write_reg(ov5693->client, OV5693_8BIT, OV5693_SW_STREAM,
+			       standby ? OV5693_STOP_STREAMING : OV5693_START_STREAMING);
 
-	ret = regulator_bulk_enable(OV5693_NUM_SUPPLIES,
-			sensor->supplies);
-	if (ret)
-		goto fail_power;
+	return ret;
```

Comment:
Ack! I realised this was stupid and fixed it in the next commit (which still needs a bit more work), but since this will need changes anyway I'll update it here.